sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/ConsumerSpEL.java

```diff
@@ -57,4 +74,31 @@ public void evaluateAssign(Consumer consumer, Collection<TopicPartition> topicPa
     mapContext.setVariable("tp", topicPartitions);
     assignExpression.getValue(mapContext);
   }
+
+  private boolean hasTimestamp(){
```

Comment:
Why do you need this method? Could you check the return type when you check 'timestamp' method?